frontend/catalyst/from_plxpr/qubit_handler.py

```diff
@@ -95,16 +121,27 @@ def set(self, qreg: AbstractQreg):
         # The old qreg SSA value is no longer usable since a new one has appeared
         # Therefore all dangling qubits from the old one also all expire
         # These dangling qubit values will be dead, so there must be none.
-        if len(self.wire_map) != 0:
+        if len(self.wire_map) != 0 and self.abstract_qreg_val is not None:
             raise CompileError(
                 "Setting new qreg value, but the previous one still has dangling qubits."
             )
+
+        # Devalidate the old qubit values if user wants to set a qreg
+        if self.abstract_qreg_val is None:
+            self.wire_map = {}
+            self.qubit_indices = []
+
         self.abstract_qreg_val = qreg
```

Comment:
I think the logic here can be a bit more clarified by just cleanly separating the modes:
```suggestion
        if self.abstract_qreg_val is not None:
           # qreg mode
           if len(self.wire_map) != 0:
               raise CompileError(
                   "Setting new qreg value, but the previous one still has dangling qubits."
               )
           self.abstract_qreg_val = qreg
        else:
           # qubits mode
           # Devalidate the old qubit values if user wants to set a qreg
            self.wire_map = {}
            self.qubit_indices = []


```